pkgs/development/libraries/science/math/nccl/tests.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, which
+, cudaPackages
+, mpiSupport ? false
+, mpi
+}:
+
+assert mpiSupport -> mpi != null;
+
+with cudaPackages;
```

Comment:
```suggestion
with cudaPackages;
```

Nit: using `with` makes it hard to reason about where packages are coming from. I'd strongly recommend against it.